pkg/image/controller/controller.go

```diff
@@ -39,35 +33,105 @@ const retryCount = 2
 // the image stream is not modified (so it will be tried again later). If a permanent
 // failure occurs the image is marked with an annotation. The tags of the original spec image
 // are left as is (those are updated through status).
+// There are 3 use cases here:
+// 1. spec.DockerImageRepository defined without any tags results in all tags being imported
+//    from default registry
+// 2. spec.DockerImageRepository + tags defined - import only the provided tags,
+//    for kind == DockerImage import underlying image, iow. exact tag (no provided means latest),
+//    for kind != DockerImage import tag from the default registry
```

Comment:
Why do we need import policy?  Specifying a DockerImageRepository in
ImageStreamSpec is an effective importPolicy of "all".  If you don't want
"all", create one tag per import.

On Thu, Oct 1, 2015 at 12:14 PM, Maciej Szulik <notifications@github.com>
wrote:

> In pkg/image/controller/controller.go
> <https://github.com/openshift/origin/pull/4853#discussion_r40898234>:
>
> > @@ -39,35 +33,105 @@ const retryCount = 2
> >  // the image stream is not modified (so it will be tried again later). If a permanent
> >  // failure occurs the image is marked with an annotation. The tags of the original spec image
> >  // are left as is (those are updated through status).
> > +// There are 3 use cases here:
> > +// 1. spec.DockerImageRepository defined without any tags results in all tags being imported
> > +//    from default registry
> > +// 2. spec.DockerImageRepository + tags defined - import only the provided tags,
> > +//    for kind == DockerImage import underlying image, iow. exact tag (no provided means latest),
> > +//    for kind != DockerImage import tag from the default registry
>
> I wasn't specific enough on the trello, the ImageStreamSpec will have
> additional ImportPolicy field, like @ncdc <https://github.com/ncdc>
> suggested that will by default will import all, and you can change it to
> import specified.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4853/files#r40898234>.
>